File: POS Professional/POS Professional/WholesalerReports.vb
Imports System.IO
Imports System.Data.SqlClient

Public Class WholesalerReports

    Dim currentpath As String = Directory.GetCurrentDirectory()

    Private Sub cmdPreview_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdPreview.Click

        If rdAllSaler.Checked = True Then
            RptUser = "ALL"
        End If
        If rdSelSaler.Checked = True Then
            If cmbCustName.Text = "" Then gblMessage = "Invalid wholesaler selection." : MsgStatus.ShowDialog(Me) : Exit Sub
            RptUser = Mid(cmbCustName.Text.Trim, 1, 4)
        End If
        FromDate = DateFrom.Value
        If rdWDetail.Checked = True Then gblReportOption = "WDETAIL" : gblReportName = currentpath & "\" & "Reports\WholesalerDetail.rpt"
        If rdWLedger.Checked = True Then gblReportOption = "WLEDGER" : gblReportName = currentpath & "\" & "Reports\WholesalerLedger.rpt"
        Reports.MdiParent = Main
        Reports.Show()

    End Sub

    Private Sub cmdClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdClose.Click
        SqlDataSet.Clear()
        Me.Close()
        WholesalersMenu.WindowState = FormWindowState.Normal
    End Sub

    Private Sub WholesalerReports_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        StockMenu.WindowState = FormWindowState.Normal
    End Sub

    Private Sub WholsalerReports_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Call GetConnectionString()
        rdAllSaler.Checked = True
        rdWDetail.Checked = True
        Dim FillCombo As String = "SELECT * FROM SUPPLIERS ORDER BY SUPL_CODE"
        ExecuteAdapterQuery(SqlDataAdapter, FillCombo, Connection)
        Dim Reader As SqlDataReader = SqlDataAdapter.SelectCommand.ExecuteReader
        If Reader.HasRows Then
            While Reader.Read
                cmbCustName.Items.Add(Reader.Item("SUPL_CODE").ToString & "-" & Reader.Item("SUPL_NAME").ToString)
            End While
            If Not Reader Is Nothing Then Reader.Close()
        End If
    End Sub

    Private Sub rdSelSaler_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles rdSelSaler.KeyDown
        If e.KeyCode = Keys.Enter Then
            If rdSelSaler.Checked = True Then
                cmbCustName.Enabled = True
            Else
                cmbCustName.Enabled = False
            End If
        End If
    End Sub

    Private Sub rdSelSaler_MouseClick(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles rdSelSaler.MouseClick
        If rdSelSaler.Checked = True Then
            cmbCustName.Enabled = True
        Else
            cmbCustName.Enabled = False
        End If
    End Sub

    Private Sub rdAllSaler_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles rdAllSaler.KeyDown
        If e.KeyCode = Keys.Enter Then
            If rdAllSaler.Checked = True Then
                cmbCustName.Enabled = False
            Else
                cmbCustName.Enabled = True
            End If
        End If
    End Sub

    Private Sub rdAllSaler_MouseClick(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles rdAllSaler.MouseClick
        If rdAllSaler.Checked = True Then
            cmbCustName.Enabled = False
        Else
            cmbCustName.Enabled = True
        End If
    End Sub
End Class

File: POS System - Retaurant - 2022/POS System/WholesalerReports.cs
using System;
using System.IO;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace POS_System
{

    public partial class WholesalerReports
    {

        private string currentpath = Directory.GetCurrentDirectory();

        public WholesalerReports()
        {
            InitializeComponent();
        }

        private void cmdPreview_Click(object sender, EventArgs e)
        {

            if (rdAllSaler.Checked == true)
            {
                mdlGeneral.RptUser = "ALL";
            }
            if (rdSelSaler.Checked == true)
            {
                if (string.IsNullOrEmpty(cmbCustName.Text))
                {
                    mdlGeneral.gblMessage = "Invalid wholesaler selection.";
                    My.MyProject.Forms.MsgStatus.ShowDialog(this);
                    return;
                }
                mdlGeneral.RptUser = Strings.Mid(cmbCustName.Text.Trim(), 1, 4);
            }
            mdlGeneral.FromDate = DateFrom.Value;
            if (rdWDetail.Checked == true)
            {
                mdlGeneral.gblReportOption = "WDETAIL";
                mdlGeneral.gblReportName = currentpath + @"\" + @"Reports\WholesalerDetail.rpt";
            }
            if (rdWLedger.Checked == true)
            {
                mdlGeneral.gblReportOption = "WLEDGER";
                mdlGeneral.gblReportName = currentpath + @"\" + @"Reports\WholesalerLedger.rpt";
            }
            My.MyProject.Forms.Reports.MdiParent = My.MyProject.Forms.Main;
            My.MyProject.Forms.Reports.Show();

        }

        private void cmdClose_Click(object sender, EventArgs e)
        {
            mdlGeneral.SqlDataSet.Clear();
            Close();
        }

        private void frmWholsalerReports_Load(object sender, EventArgs e)
        {
            mdlGeneral.GetConnectionString();
            rdAllSaler.Checked = true;
            rdWDetail.Checked = true;
            string FillCombo = "SELECT * FROM SUPPLIERS ORDER BY SUPL_CODE";
            mdlGeneral.ExecuteAdapterQuery(mdlGeneral.SqlDataAdapter, FillCombo, mdlGeneral.Connection);
            var Reader = mdlGeneral.SqlDataAdapter.SelectCommand.ExecuteReader();
            if (Reader.HasRows)
            {
                while (Reader.Read())
                    cmbCustName.Items.Add(Reader["SUPL_CODE"].ToString() + "-" + Reader["SUPL_NAME"].ToString());
            }
        }

        private void rdSelSaler_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                if (rdSelSaler.Checked == true)
                {
                    cmbCustName.Enabled = true;
                }
                else
                {
                    cmbCustName.Enabled = false;
                }
            }
        }

        private void rdSelSaler_MouseClick(object sender, MouseEventArgs e)
        {
            if (rdSelSaler.Checked == true)
            {
                cmbCustName.Enabled = true;
            }
            else
            {
                cmbCustName.Enabled = false;
            }
        }

        private void rdAllSaler_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                if (rdAllSaler.Checked == true)
                {
                    cmbCustName.Enabled = false;
                }
                else
                {
                    cmbCustName.Enabled = true;
                }
            }
        }

        private void rdAllSaler_MouseClick(object sender, MouseEventArgs e)
        {
            if (rdAllSaler.Checked == true)
            {
                cmbCustName.Enabled = false;
            }
            else
            {
                cmbCustName.Enabled = true;
            }
        }
    }
}